app/src/main/java/org/wikipedia/database/AppDatabase.kt

```diff
@@ -184,10 +184,36 @@ abstract class AppDatabase : RoomDatabase() {
                 database.execSQL("ALTER TABLE HistoryEntry ADD COLUMN description TEXT NOT NULL DEFAULT ''")
             }
         }
+        val MIGRATION_26_27 = object : Migration(26, 27) {
+            override fun migrate(database: SupportSQLiteDatabase) {
+                // Rename the existing HistoryEntry table, which we're preserving for now (in case
+                // things go wrong with migrations in the field).
+                database.execSQL("ALTER TABLE HistoryEntry RENAME TO HistoryEntry_old")
```

Comment:
Not sure if we should do a `DROP HistoryEntry_old` before running `ALTER TABLE`?

I guess this only happens to us because we install old/new versions back and forth, but not sure if should we do a drop table before altering a table.

![Screenshot_20250312_161339_Wikipedia Dev](https://github.com/user-attachments/assets/40bd3e48-e8ee-4182-bd2a-b8f63d092422)

Reply:
Yes it is. We can have some discussion tomorrow about this.